syntax = "proto3";

package calculator;

message BinaryOperation {
  float first_operand = 1;
  float second_operand = 2;
  Operation operation = 3;
}

message CalculationResult {
  float result = 1;
}

enum Operation {
  ADD = 0;
  SUBTRACT = 1;
}

service Calculator {
  rpc Calculate (BinaryOperation) returns (CalculationResult);
}